src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Helpers/RemoveUnnecessaryImports/AbstractUnnecessaryImportsProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.RemoveUnnecessaryImports
+{
+    internal abstract class AbstractUnnecessaryImportsProvider<T>
```

Comment:
Existing code just split out from [AbstractRemoveUnnecessaryImportsService](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsService.cs,1a00d47c7e649b76) into this provider.